apps/gdaltindex_lib.cpp

```diff
@@ -1167,218 +1366,123 @@ GDALTileIndexOptionsNew(char **papszArgv,
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
-            psOptions->osColorInterp = papszArgv[++iArg];
-        }
-        else if (EQUAL(papszArgv[iArg], "-mask"))
-        {
-            psOptions->bMaskBand = true;
-        }
-        else if (EQUAL(papszArgv[iArg], "-gti_filename"))
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
-            psOptions->osGTIFilename = papszArgv[++iArg];
-        }
-        else if (EQUAL(papszArgv[iArg], "-overwrite"))
-        {
-            psOptions->bOverwrite = true;
-        }
-        else if (EQUAL(papszArgv[iArg], "-recursive"))
-        {
-            psOptions->bRecursive = true;
-        }
-        else if (EQUAL(papszArgv[iArg], "-min_pixel_size"))
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
-            psOptions->dfMinPixelSize = CPLAtofM(papszArgv[++iArg]);
-        }
-        else if (EQUAL(papszArgv[iArg], "-max_pixel_size"))
+    }
+
+    try
+    {
+        auto argParser = GDALTileIndexAppOptionsGetParser(psOptions.get(),
+                                                          psOptionsForBinary);
+        argParser->parse_args_without_binary_name(aosArgv.List());
+
+        // Check all no store_into args
+        if (auto oTr = argParser->present<std::vector<double>>("-tr"))
         {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
-            psOptions->dfMaxPixelSize = CPLAtofM(papszArgv[++iArg]);
+            psOptions->xres = (*oTr)[0];
+            psOptions->yres = (*oTr)[1];
         }
-        else if (EQUAL(papszArgv[iArg], "-filename_filter"))
+
+        if (auto oTe = argParser->present<std::vector<double>>("-te"))
         {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
-            psOptions->oSetFilenameFilters.insert(papszArgv[++iArg]);
+            psOptions->xmin = (*oTe)[0];
+            psOptions->ymin = (*oTe)[1];
+            psOptions->xmax = (*oTe)[2];
+            psOptions->ymax = (*oTe)[3];
         }
-        else if (EQUAL(papszArgv[iArg], "-fetch_md"))
+
+        if (auto fetchMd =
+                argParser->present<std::vector<std::string>>("-fetch_md"))
         {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(3);
-            GDALTileIndexRasterMetadata md;
-            md.osRasterItemName = papszArgv[++iArg];
-            md.osFieldName = papszArgv[++iArg];
-            const char *pszType = papszArgv[++iArg];
-            if (EQUAL(pszType, "String"))
-                md.eType = OFTString;
-            else if (EQUAL(pszType, "Integer"))
-                md.eType = OFTInteger;
-            else if (EQUAL(pszType, "Integer64"))
-                md.eType = OFTInteger64;
-            else if (EQUAL(pszType, "Real"))
-                md.eType = OFTReal;
-            else if (EQUAL(pszType, "Date"))
-                md.eType = OFTDate;
-            else if (EQUAL(pszType, "DateTime"))
-                md.eType = OFTDateTime;
-            else
+            if (fetchMd->size() % 3)
```

Comment:
is this check really needed ? argparse constraints should have validated that already, shouldn't they ? Or perhaps make it a CPLAssert() ?

Reply:
no, it's a leftover for when I thought it supported multiple 3*n arguments. I'll remove it.

Reply:
done (CPLAssert)